File: CS/MVVMDemo.Wpf/Modules/Services/CustomServiceViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using System.Diagnostics;
using System.IO;
using System.Windows;

namespace MVVMDemo.Services {
    public class CustomServiceViewModel {
        public void Export(ExportFormat format) {
            var path = Path.ChangeExtension(Path.GetTempFileName(), format.ToString().ToLower());

            IExportService service = this.GetRequiredService<IExportService>();
            using(FileStream file = File.Create(path)) {
                service.Export(file, format);
            }

            if(MessageBox.Show("Do you want to open the file?\r\n" + path, "Open file", MessageBoxButton.YesNo) == MessageBoxResult.Yes)
                Process.Start(path);
            else
                File.Delete(path);
        }
    }
    public interface IExportService {
        void Export(Stream stream, ExportFormat format);
    }
    public enum ExportFormat {
        Xlsx, Pdf
    }
}

File: VB/MVVMDemo.Wpf/Modules/Services/CustomServiceViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports System.Diagnostics
Imports System.IO
Imports System.Windows

Namespace MVVMDemo.Services
    Public Class CustomServiceViewModel
        Public Sub Export(ByVal format As ExportFormat)
            Dim path = System.IO.Path.ChangeExtension(System.IO.Path.GetTempFileName(), format.ToString().ToLower())

            Dim service As IExportService = Me.GetRequiredService(Of IExportService)()
            Using file As FileStream = System.IO.File.Create(path)
                service.Export(file, format)
            End Using

            If MessageBox.Show("Do you want to open the file?" & ControlChars.CrLf & path, "Open file", MessageBoxButton.YesNo) = MessageBoxResult.Yes Then
                Process.Start(path)
            Else
                System.IO.File.Delete(path)
            End If
        End Sub
    End Class
    Public Interface IExportService
        Sub Export(ByVal stream As Stream, ByVal format As ExportFormat)
    End Interface
    Public Enum ExportFormat
        Xlsx
        Pdf
    End Enum
End Namespace
